packages/transit-vehicle-overlay/src/index.tsx

```diff
@@ -37,29 +79,39 @@ const TransitVehicleOverlay = ({
     return null;
   }
 
-  const Tooltip = TooltipSlot || VehicleTooltip;
-
-  return (
-    <>
-      {validVehicles.map(vehicle => {
-        const Icon = getTransitIcon(vehicle.routeType);
-
-        return (
-          <MarkerWithPopup
-            key={vehicle.vehicleId}
-            position={[vehicle.lat, vehicle.lon]}
-            // @ts-expect-error the prop override doesn't require all props to be present
-            popupProps={{ offset: [-15, 0] }}
-            // @ts-expect-error TODO FIX
-            tooltipContents={<Tooltip vehicle={vehicle} />}
-          >
-            {/* @ts-expect-error We know the icon is set dynamically */}
-            <Icon rotate={vehicle.heading} routeColor={color} />
-          </MarkerWithPopup>
-        );
-      })}
-    </>
-  );
+  return validVehicles?.map(vehicle => (
+    <MarkerWithPopup
+      key={vehicle.vehicleId}
+      // @ts-expect-error the prop override doesn't require all props to be present
+      popupProps={{ offset: [-iconPixels / 2 - iconPadding, 0] }}
+      position={[vehicle.lat, vehicle.lon]}
+      tooltipContents={
+        vehicle.routeShortName && <TooltipSlot vehicle={vehicle} />
+      }
+    >
+      <IconContainer
+        padding={iconPadding}
+        pixels={iconPixels}
+        vehicle={vehicle}
+      >
+        <VehicleIcon
+          defaultMode={defaultMode}
+          ModeIcon={ModeIcon}
+          vehicle={vehicle}
+        />
+      </IconContainer>
+    </MarkerWithPopup>
+  ));
 };
 
 export default TransitVehicleOverlay;
+
+// Export the other subcomponents
+export {
+  CircleWithCaret,
+  DefaultVehicleIcon,
+  RotatingCircle,
+  RouteNumberIcon,
+  withRouteColorBackground,
+  withRouteColorBackgroundOnHover
```

Comment:
Does all of this need to be exported? Are we planning on using it elsewhere?

Reply:
Yes, that's for use in OTP-RR!